### PR TITLE
Avoid using go routines in reconcile function

### PR DESCRIPTION
**Pull request by shirmoran**
This PR avoids the use of go routines in the reconcile loop in order to maximize the efficiency of the controller.

**The updated Reconcile loop works as follow:**

image

When checking if the timeout of the certsuite container has been exceeded, 2 minutes are added to the given time out as a safeguard.

In addition, this PR removes an unnecessary check whether the run CR has been updated, as we use an event filter to avoid calling the reconcile function in those cases.

Also, some unnecessary functions were removed, as well as their corresponding unit tests.

### CODE REVIEW

**Review by sebrandon1 — APPROVED** [review]

File: internal/controller/certsuiterun_controller.go [hunk 1]
```diff
@@ -132,72 +133,64 @@ func getJobRunTimeThreshold(timeoutStr string) time.Duration {
...
+	certSuiteContainer := getContainerStatus(&certSuitePod, definitions.CnfCertSuiteContainerName)
+	if certSuiteContainer == nil {
+		return ctrl.Result{}, fmt.Errorf("container %s does not exist in pod %s", definitions.CnfCertSuiteContainerName, certSuitePodNamespacedName)
+	}
+	if certSuiteContainer.State.Terminated == nil {
+		// check whether timeout has exceeded.
+		elapsedTime := time.Since(certSuiteContainer.State.Running.StartedAt.Time)
+		certSuiteTimeout := getJobRunTimeThreshold(timeout)
+		if elapsedTime > certSuiteTimeout+certSuiteTimeoutSafeGuard {
+			return ctrl.Result{}, fmt.Errorf("timeout of %s pod has reached while pod is still running", certSuitePodNamespacedName)
```
[the remark this answers is not in the document]
Comment (shirmoran) [reply internal/controller/certsuiterun_controller.go:170]: I think you are referring to phase and not verdict, right?

[the remark this answers is not in the document]
Comment (greyerof) [reply internal/controller/certsuiterun_controller.go:170]: Yes, you're right. Verdict should be set to StatusVerdictFailed but we need to create a new status phase "StatusPhaseCertSuiteTimeout" for this.

[the remark this answers is not in the document]
Comment (shirmoran) [reply internal/controller/certsuiterun_controller.go:170]: Maybe `StatusVerdictError` for the verdict? as the tests did not actually fail

[the remark this answers is not in the document]
Comment (greyerof) [reply internal/controller/certsuiterun_controller.go:170]: Both work for me, as long as the possible combinations of certsuite outcomes that result on that verdict value are well documented.

**Review by greyerof — APPROVED** [review]

**Pull request merged**